### PR TITLE
SONARIAC-1175 Add support for already commented lines in "CommentLocationVisitor"

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/CommentLocationVisitor.java [hunk 1]
```diff
@@ -90,9 +91,22 @@ public void visitComment(InputFileContext ctx, YamlTree tree) {
 
   private void processComment(InputFileContext ctx, Comment comment) {
     if (IS_LINE_NUMBER.matcher(comment.contentText()).matches()) {
+      // there is only line comment
       int lineCommentLocation = comment.textRange().start().line();
       var lineCommentValue = Integer.parseInt(comment.contentText());
       shifter.addShiftedLine(ctx, lineCommentLocation, lineCommentValue);
+    } else {
+      var matcher = CONTAINS_LINE_NUMBER.matcher(comment.contentText());
+      if (matcher.find()) {
```
Comment (jonas-wielage-sonarsource) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/CommentLocationVisitor.java:100]: Nice approach how you match it (using combination of `.find()` and `$`.)

**Review by jonas-wielage-sonarsource — APPROVED** [review]
LGTM, please see my suggestion

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/CommentLocationVisitor.java (hunk 1 above)
Comment (mstachniuk) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/CommentLocationVisitor.java:100]: Thanks for this comment. It reminds me about one another edge case

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/CommentLocationVisitor.java [hunk 2]
```diff
@@ -89,10 +89,15 @@ public void visitComment(InputFileContext ctx, YamlTree tree) {
...
+    var matcher = CONTAINS_LINE_NUMBER.matcher(comment.value());
+    if (matcher.find()) {
       int lineCommentLocation = comment.textRange().start().line();
-      var lineCommentValue = Integer.parseInt(comment.contentText());
+      var lineCommentValue = Integer.parseInt(matcher.group("number"));
       shifter.addShiftedLine(ctx, lineCommentLocation, lineCommentValue);
+    } else {
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Line number comment not detected, comment: {}", comment.value());
+      }
```
Comment (nils-werner-sonarsource) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/CommentLocationVisitor.java:100]: The condition is implicit in the debug() call. This is bad code IMO.

**Pull request merged**